### PR TITLE
[test] Optimize test generation tool

### PR DESCRIPTION
**Pull request by Honry**
- Centeralize integer arithmetic ops in one place
- Remove LaneNumber class as which is duplicated with simd_lane_value.py
- No impact for generated tests

Pulled from https://github.com/WAVM/WAVM/pull/268

### CODE REVIEW

**Comment by Honry** [comment]
@tlively, @arunetm, PTAL, thanks!

File: test/core/simd/meta/simd_integer_op.py [hunk 1]
```diff
@@ -5,98 +5,184 @@
...
+                value = p1 + p2
+            if self.op.startswith('sub'):
+                value = p1 - p2
+
+            if value > lane.mask:
+                return lane.mask
+            if value < 0:
+                return 0
+
+        return value
```
Comment (tlively) [inline test/core/simd/meta/simd_integer_op.py:90]: If I understand correctly, this return should never be reached. Would it make more sense to make it an error?

File: test/core/simd/meta/simd_integer_op.py [hunk 2]
```diff
@@ -5,98 +5,184 @@
...
+        :param p: the operand, integer or literal string in hex or decimal format
+        :param lane: the LaneValue instance of a lane in v128
+        :return: the string of the result of <self.op p> in hex or decimal format
+        """
+        v = p
+        base = 10
+        if isinstance(p, str):
+            if '0x' in p:
+                base = 16
+            v = int(p, base)
```
[the remark this answers is not in the document]
Comment (Honry) [reply test/core/simd/meta/simd_integer_op.py:107]: Introduced a variable `v` here was intended to return `p` without any conversion in the `abs` operation when `p` is positive. If we re-use `p` here, we might not get the original format back. 

and yes, `p` is the `operand`. I'll improve the name.

File: test/core/simd/meta/simd_integer_op.py [hunk 3]
```diff
@@ -5,98 +5,184 @@
...
-        v = int(p, base)
+        value &= lane.mask
+        if signed:
+            if value > lane.max:
+                return value - lane.mod
+            if value < lane.min:
+                return value + lane.mod
+        return value
+
+    def saturate(self, p1: int, p2: int, lane: LaneValue) -> int:
```
[the remark this answers is not in the document]
Comment (Honry) [reply test/core/simd/meta/simd_integer_op.py:44]: Sure, good catch, thanks!

File: test/core/simd/meta/simd_integer_op.py (hunk 1 above)
Comment (Honry) [reply test/core/simd/meta/simd_integer_op.py:90]: Actually, this `return` statement can be reached when the value is in a normal scope. The branches above can only handle the values out of boundary values.

**Comment by Honry** [comment]
@tlively, thanks for your review! Some comments addressed, PTAL, thanks!

**Comment by tlively** [comment]
Great, thank you!
